src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -0,0 +1,222 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Centipede engine interface."""
+
+from pathlib import Path
+import re
+
+from clusterfuzz._internal.bot.fuzzers import dictionary_manager
+from clusterfuzz._internal.bot.fuzzers import engine_common
+from clusterfuzz._internal.bot.fuzzers import utils as fuzzer_utils
+from clusterfuzz._internal.metrics import logs
+from clusterfuzz._internal.system import environment
+from clusterfuzz._internal.system import new_process
+from clusterfuzz.fuzz import engine
+
+_CLEAN_EXIT_SECS = 10
+_TIMEOUT = 25
+_SERVER_COUNT = 1
+_RSS_LIMIT = 4096
+_RLIMIT_AS = 5120
+_ADDRESS_SPACE_LIMIT = 0
+_DEFAULT_ARGUMENTS = [
+    '--exit_on_crash=1',
+    f'--timeout={_TIMEOUT}',
+    f'--fork_server={_SERVER_COUNT}',
+    f'--rss_limit_mb={_RSS_LIMIT}',
+    f'--address_space_limit_mb={_ADDRESS_SPACE_LIMIT}',
+]
+
+_CRASH_REGEX = re.compile(r'Crash detected, saving input to (.*)')
+
+
+class CentipedeError(Exception):
+  """Base exception class."""
+
+
+def _get_runner():
+  """Gets the Centipede runner."""
+  centipede_path = Path(environment.get_value('BUILD_DIR'), 'centipede')
+  if not centipede_path.exists():
+    raise CentipedeError('Centipede not found in build')
+
+  centipede_path = str(centipede_path)
+  if environment.get_value('USE_UNSHARE'):
+    return new_process.UnicodeModifierRunner(centipede_path)
+  return new_process.UnicodeProcessRunner(centipede_path)
+
+
+def _get_reproducer_path(log):
+  """Gets the reproducer path, if any."""
+  crash_match = _CRASH_REGEX.search(log)
+  if not crash_match:
+    return None
+
+  return crash_match.group(1)
+
+
+class Engine(engine.Engine):
+  """Centipede engine implementation."""
+
+  @property
+  def name(self):
+    return 'centipede'
+
+  def prepare(self, corpus_dir, target_path, build_dir):  # pylint: disable=unused-argument
+    """Prepares for a fuzzing session, by generating options.
+
+    Args:
+      corpus_dir: The main corpus directory.
+      target_path: Path to the target.
+      build_dir: Path to the build directory.
+
+    Returns:
+      A FuzzOptions object.
+    """
+    arguments = []
+    dict_path = Path(
+        dictionary_manager.get_default_dictionary_path(target_path))
+    if dict_path.exists():
+      arguments.append(f'--dictionary={dict_path}')
+
+    # Directory workdir saves:
+    # 1. Centipede-readable corpus file;
+    # 2. Centipede-readable feature file;
+    # 3. Crash reproducing inputs.
+    workdir = Path(corpus_dir).parent / 'workdir'
```

Comment:
We don't need this. Can we just have `self._create_temp_dir` return the full created path, and just pass `'workdir'` to it? 

i.e.

```python
workdir = self._create_temp_dir('workdir')
```

Reply:
Ah sorry I miss interpreted your comment earlier, fixing now : )